diff --git a/include/openssl/base.h b/include/openssl/base.h
index 16796c5..2e70f20 100644
--- a/include/openssl/base.h
+++ b/include/openssl/base.h
@@ -100,6 +100,8 @@ extern "C" {
 
 #define AWSLC_VERSION_NAME "AWS-LC"
 #define OPENSSL_IS_AWSLC
+// compatible with nginx
+#define OPENSSL_IS_BORINGSSL
 // |OPENSSL_VERSION_NUMBER| should match the version number in opensslv.h.
 #define OPENSSL_VERSION_NUMBER 0x1010107f
 #define SSLEAY_VERSION_NUMBER OPENSSL_VERSION_NUMBER
diff --git a/include/openssl/ssl.h b/include/openssl/ssl.h
index 7208c4b..81de8e1 100644
--- a/include/openssl/ssl.h
+++ b/include/openssl/ssl.h
@@ -6452,4 +6452,9 @@ BSSL_NAMESPACE_END
 #define SSL_R_TLSV1_ALERT_ECH_REQUIRED 1121
 #define SSL_R_SERIALIZATION_INVALID_SERDE_VERSION 1122
 
+// ref: https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_ssl_ocsp.c#L514
+#define SSL_CTRL_GET_TLSEXT_STATUS_REQ_TYPE     127
+// ref: https://github.com/freenginx/nginx/blob/default/src/event/quic/ngx_event_quic_openssl_compat.h#L10
+#define SSL_R_MISSING_QUIC_TRANSPORT_PARAMETERS_EXTENSION 801
+
 #endif  // OPENSSL_HEADER_SSL_H
