diff --git a/include/openssl/base.h.in b/include/openssl/base.h.in
index 60877e9..0c2d2ff 100644
--- a/include/openssl/base.h.in
+++ b/include/openssl/base.h.in
@@ -100,6 +100,8 @@ extern "C" {
 
 #define AWSLC_VERSION_NAME "AWS-LC"
 #define OPENSSL_IS_AWSLC
+// compatible with nginx
+#define OPENSSL_IS_BORINGSSL
 // |OPENSSL_VERSION_NUMBER| should match the version number in opensslv.h.
 #define OPENSSL_VERSION_NUMBER @OPENSSL_VERSION_NUMBER@
 #define SSLEAY_VERSION_NUMBER OPENSSL_VERSION_NUMBER
diff --git a/include/openssl/ssl.h b/include/openssl/ssl.h
index 0e75768..50f94c0 100644
--- a/include/openssl/ssl.h
+++ b/include/openssl/ssl.h
@@ -6217,6 +6217,10 @@ OPENSSL_EXPORT OPENSSL_DEPRECATED int SSL_set_tmp_rsa(SSL *ssl, const RSA *rsa);
 #define SSL_CTRL_GET_SESS_CACHE_MODE doesnt_exist
 #define SSL_CTRL_GET_SESS_CACHE_SIZE doesnt_exist
 #define SSL_CTRL_GET_TLSEXT_STATUS_REQ_TYPE doesnt_exist
+// ref: https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_ssl_ocsp.c#L514
+//#define SSL_CTRL_GET_TLSEXT_STATUS_REQ_TYPE     127
+// ref: https://github.com/freenginx/nginx/blob/default/src/event/quic/ngx_event_quic_openssl_compat.h#L10
+#define SSL_R_MISSING_QUIC_TRANSPORT_PARAMETERS_EXTENSION 801
 #define SSL_CTRL_GET_TLSEXT_TICKET_KEYS doesnt_exist
 #define SSL_CTRL_GET_TOTAL_RENEGOTIATIONS doesnt_exist
 #define SSL_CTRL_MODE doesnt_exist
