diff --git a/src/event/ngx_event_openssl.c b/src/event/ngx_event_openssl.c
index e36f30c74..4346ebf17 100644
--- a/src/event/ngx_event_openssl.c
+++ b/src/event/ngx_event_openssl.c
@@ -793,7 +793,7 @@ ngx_ssl_client_certificate(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *cert,
         }
 
 #ifdef OPENSSL_IS_BORINGSSL
-        if (sk_X509_NAME_find(list, NULL, name) > 0) {
+        if (sk_X509_NAME_find_awslc(list, NULL, name) > 0) {
 #else
         if (sk_X509_NAME_find(list, name) >= 0) {
 #endif
diff --git a/src/event/quic/ngx_event_quic.h b/src/event/quic/ngx_event_quic.h
index bab085f46..a189dcebe 100644
--- a/src/event/quic/ngx_event_quic.h
+++ b/src/event/quic/ngx_event_quic.h
@@ -12,7 +12,10 @@
 #include <ngx_core.h>
 
 
-#if (OPENSSL_VERSION_NUMBER >= 0x30500010L)
+#if (defined OPENSSL_IS_AWSLC)
+#define NGX_QUIC_BORINGSSL_API               1
+
+#elif (OPENSSL_VERSION_NUMBER >= 0x30500010L)
 #define NGX_QUIC_OPENSSL_API                 1
 
 #elif (defined SSL_R_MISSING_QUIC_TRANSPORT_PARAMETERS_EXTENSION)
