diff --git a/config b/config
index a621403..e7cde6c 100644
--- a/config
+++ b/config
@@ -159,7 +159,7 @@ ngx_module_libs=
 if [ $ngx_found = yes ]; then
     # this is a hack to persuade nginx's build system to favor
     # the paths set by our user environment
-    CFLAGS="$ngx_lua_opt_I $CFLAGS"
+    CFLAGS="$ngx_lua_opt_I $CFLAGS -Wno-error=unused-variable -Wno-error=unused-function -Wno-error=conditional-uninitialized"
     NGX_LD_OPT="$ngx_lua_opt_L $NGX_LD_OPT"
 
     ngx_module_incs="$ngx_module_incs $ngx_feature_path"
diff --git a/src/ngx_stream_lua_proxy_ssl_verifyby.c b/src/ngx_stream_lua_proxy_ssl_verifyby.c
index 99c9ce6..f6cd6fa 100644
--- a/src/ngx_stream_lua_proxy_ssl_verifyby.c
+++ b/src/ngx_stream_lua_proxy_ssl_verifyby.c
@@ -801,13 +801,13 @@ ngx_stream_lua_ffi_proxy_ssl_get_verify_cert(ngx_stream_lua_request_t *r,
 
     *err = "LibreSSL does not support this function";
 
-    return NGX_ERROR;
+    return NULL;
 
 #elif defined(OPENSSL_IS_BORINGSSL)
 
     *err = "BoringSSL does not support this function";
 
-    return NGX_ERROR;
+    return NULL;
 
 #else
 
diff --git a/src/ngx_stream_lua_ssl_client_helloby.c b/src/ngx_stream_lua_ssl_client_helloby.c
index db478ec..b17ac32 100644
--- a/src/ngx_stream_lua_ssl_client_helloby.c
+++ b/src/ngx_stream_lua_ssl_client_helloby.c
@@ -218,7 +218,7 @@ ngx_stream_lua_ssl_client_hello_handler(ngx_ssl_conn_t *ssl_conn,
         return -1;
     }
 
-#if (nginx_version > 1029001)
+#if (nginx_version > 1029001 && defined(SSL_CLIENT_HELLO_SUCCESS))
     /* see commit 0373fe5d98c1515640 for more details */
     rc = ngx_ssl_client_hello_callback(ssl_conn, al, arg);
 
