diff --git a/src/http/ngx_http_request.c b/src/http/ngx_http_request.c
index aeaac77..0f86952 100644
--- a/src/http/ngx_http_request.c
+++ b/src/http/ngx_http_request.c
@@ -2057,7 +2057,7 @@ ngx_http_process_request(ngx_http_request_t *r)
         if (c->ssl == NULL) {
             ngx_log_error(NGX_LOG_INFO, c->log, 0,
                           "client sent plain HTTP request to HTTPS port");
-            ngx_http_finalize_request(r, NGX_HTTP_TO_HTTPS);
+            ngx_http_finalize_request(r, NGX_HTTP_CLOSE);
             return;
         }
 
